[Rahul Prasad] iP

rahulprasad01 opened the pull request:
# Duke

**Duke schedules your events, deadlines, etc and helps you organise better**

It is:
1. Text based
2. User friendly
3. Easy to modify

Here's how you can use it:
- Download it [Github pages](https://github.com/rahulprasad01/ip)
- Add and modify tasks

## It is free to use.

Feature:

- [x] Managing events
- [x] Managing deadlines
- [x] Set tasks to be done after a certain date


* * If you Java programmer, you can use it to practice Java too. Here's the Launcher method:

```
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

rahulprasad01 committed:
level 1

rahulprasad01 committed:
level 2

rahulprasad01 committed:
level 3

rahulprasad01 committed:
level 3

rahulprasad01 committed:
level 3

rahulprasad01 committed:
level 3

rahulprasad01 committed:
level 3

rahulprasad01 committed:
level 4

rahulprasad01 committed:
level 5

rahulprasad01 committed:
level 6

rahulprasad01 committed:
Add Level 7

rahulprasad01 committed:
Add Level-8

Punpun1643 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,200 @@
+        while (!input[0].equals("bye")) {
+
+try {
+    if (input[0].equals("list")) {
```
Really clean if-else blocks that are easy to understand. 👍

Punpun1643 commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,87 @@
+            savedTasks.createNewFile();
+        }
+    }
+    public void writeTasks( ) throws IOException {
```
Did you accidentally add a while space here? Minor issue but feels a bit weird 😆

shurvirarora replied:
Good spot! I think the white space may have been on accident.

Punpun1643 commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,87 @@
+                        .append(System.lineSeparator())
+                        .append(((Deadline) curr).getBy())
+                        .append(System.lineSeparator());
+            } else if (curr instanceof Event) {
+                textToAdd.append("E ")
+                        .append(curr.isDone() ? "1 " : "0 ")
+                        .append(curr.getName())
+                        .append(System.lineSeparator())
+                        .append(((Event) curr).getAt())
+                        .append(System.lineSeparator());
```
This looks really clean by separating the append on different lines

aditi2313 replied:
Agreed! Code looks very neat and consistent this way! 👍🏻

Punpun1643 commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,87 @@
+        while (sc.hasNext()) {
+            String current = sc.nextLine();
+            String[] splits = current.split(" ", 3);
+            switch (splits[0]) {
```
Please remember to follow the module's switch statements standard. 😉

shurvirarora replied:
You can manually change the indentation in intelliJ's settings too!

Punpun1643 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,44 @@
+    public void markAsDone() {
+        this.isDone = true;
+        
+        
```
Maybe can consider removing unused line spacing...same goes for some other places

Punpun1643 reviewed (commented):
Generally, really clean code. Just some minor details that I have pointed out 👍

shurvirarora commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,200 @@
+       String description = "";
+       String by;
+        StringBuilder sb = new StringBuilder();
+        for(int i = 1; i < input.length; i++){
```
White spacing is inconsistent here - just some minor fixes with Java Coding Standard should do 👍

shurvirarora commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,200 @@
+                sb.append(input[i] + " ");
+
+            }
+            else {
```
If-else spacing/indentation isn't following Java Coding Standard here.

shurvirarora commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,200 @@
+} catch (DukeException exception)  {
+    System.out.println(exception.getMessage());
+}
+   catch (IOException ie) {
```
Spacing with catch statement also can be fixed easily 👍

shurvirarora commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,44 @@
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
```
Good naming for boolean variable with use of "is" 👍

shurvirarora reviewed (commented):
Overall code looks good - just some nits regarding line indentation & white spacing 👍

rahulprasad01 committed:
A-MoreOOP

aditi2313 commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,15 @@
+import java.io.IOException;
+
+public abstract class Command {
+    private boolean exit = false;
```
Perhaps consider using a better name that sounds like a boolean for this variable?

aditi2313 commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,29 @@
+    public static Command parseCommand(String fullCommand) throws DukeException {
+        String[] first = fullCommand.split(" ", 2);
+        switch (first[0]) {
+            case ("bye"):
+                return new ExitCommand();
+            case ("list"):
+                return new ListCommand();
+            case ("todo"):
+                return new AddCommand(new Todo(first[1]));
+            case ("event"): {
+                String[] second = first[1].split("/at", 2);
+                return new AddCommand(new Event(second[0], second[1]));
+            }
+            case ("deadline"): {
+                String[] second = first[1].split("/by", 2);
+                return new AddCommand(new Deadline(second[0], (second[1])));
+            }
+            case ("done"):
+                return new DoneCommand(Integer.parseInt(first[1]));
+            case ("delete"):
+                return new DeleteCommand(Integer.parseInt(first[1]));
```
Remember to not have indentation for case clauses in switch statements :)
You can manually make this your default in intelliJ! :D

aditi2313 reviewed (commented):
Overall, your code looks pretty neat and consistent with the Java Coding Standard! There are just a few small violations here and there but I'm sure you can fix those 😁

rahulprasad01 committed:
A-JavaDoc

rahulprasad01 committed:
A-CodingStandard

rahulprasad01 committed:
A-CodingStandard

rahulprasad01 committed:
Merge commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d'

* commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d':
  Add Gradle support

rahulprasad01 committed:
no message

rahulprasad01 committed:
GUI

rahulprasad01 committed:
GUI

rahulprasad01 committed:
Added checkstyle and B-DoAfterTask

rahulprasad01 committed:
J unit tests

rahulprasad01 committed:
no message

rahulprasad01 committed:
Merge pull request #2 from rahulprasad01/branch-B-DoAfterTask

Merge branch-B-DoAfterTask to master

rahulprasad01 committed:
Fix minor bugs

rahulprasad01 committed:
Merge branch 'branch-B-DoAfterTask'

* branch-B-DoAfterTask:
  Fix minor bugs

rahulprasad01 committed:
Some modification

rahulprasad01 committed:
FindCommand: Initially could only search for a keyword
Modified to be able to search for multiple keywords using the java varags feature.
This can help avoid multiple searches

rahulprasad01 committed:
Merge branch 'master' into branch-A-varag

rahulprasad01 committed:
Merge pull request #3 from rahulprasad01/branch-A-varag

FindCommand: Initially could only search for a keyword

rahulprasad01 committed:
Added Minor modification

rahulprasad01 committed:
Added packaging
Organised into packages for easier access.

rahulprasad01 committed:
no message

rahulprasad01 committed:
Update README.md

rahulprasad01 committed:
Added some bug fixes

rahulprasad01 committed:
Added Ui.png

rahulprasad01 committed:
Update README.md

rahulprasad01 committed:
Update README.md

rahulprasad01 committed:
Set theme jekyll-theme-cayman

rahulprasad01 committed:
Added shadow jar plugin

rahulprasad01 committed:
Bug fixes

rahulprasad01 committed:
Bug fixes

rahulprasad01 commented:
# Duke

**Duke schedules your events, deadlines, etc and helps you organise better**

It is:
1. Text based
2. User friendly
3. Easy to modify

Here's how you can use it:
- Download it [Github pages](https://github.com/rahulprasad01/ip)
- Add and modify tasks

## It is free to use.

Feature:

- [x] Managing events
- [x] Managing deadlines
- [x] Set tasks to be done after a certain date


* * If you Java programmer, you can use it to practice Java too. Here's the Launcher method:

```
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```
